File: 프로그래머스/lv2/권민서/카펫.java
import java.util.*;

class Solution {

    public int[] solution(int brown, int yellow) {
        List<Integer> yrow = new ArrayList<>();
        int[] answer = new int[2];


        for (int i = 1; i < yellow; i++) {
            if(yellow/i < i) break;
            if (yellow % i == 0) {
                yrow.add(yellow/i);
            }
        }

        if(yrow.isEmpty()) yrow.add(1);

        Collections.sort(yrow);

        for (Integer x : yrow) {
            int row = x;
            int col = yellow/x;
            int tmp = brown;

            tmp -= 2*x;
            col += 2;
            if (tmp % 2 == 0 && tmp / 2 == col) {
                answer[0] = row + 2;
                answer[1] = col;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv2/이하은/카펫.kt
/**
1 2 3 4 
5 6 7 8
9 10 11 12

brown = 2 * w + 2 * (h - 2) = 10
yellow = (w * h) - brown = 2

w + h = (brown / 2) + 2 = 7
w * h = yellow + brown = 12
w >= h (최소 3)

*/
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var h = 3
        var w = (brown / 2) + 2 - h
        
        while(w * h != yellow + brown){
            w--
            h++
        }
        
        return intArrayOf(w, h)
    }
}